Bullwinkle.class.nut

```diff
@@ -561,9 +595,7 @@ class Bullwinkle.Package {
             local d = date();
             return format("%d.%06d", d.time, d.usec);
         } else {
-            local d = math.abs(hardware.micros());
-            return format("%d.%06d", d/1000000, d%1000000);
+            return format("%d.%06d", time(), hardware.micros());  //this can be a bit of an ugly _ts but it allows us to calculate latencies up to 36 minutes long...
```

Comment:
Confirmed, it is seriously ugly. You have littered the code with split() calls by doing this. I suggest having `_ts` as an array so you don't have to keep splitting it or make a function to split it like you have a function to create it.

The problem is it is NOT a valid use of hardware.micros(). On the agent it is valid, on the device it is not. hardware.micros() is not a sub-value of time().